### PR TITLE
Task-35817: Add Cloud Storage page

### PR DESCRIPTION
**Pull request by AzmiTouil**
Add a new page "Cloud Storage" under the Documents section , allowing to enable/disable Cloud drives from the platform.

### CODE REVIEW

File: ecm-wcm-extension/src/main/webapp/attachments-selector/attachmentsService.js [hunk 1]
```diff
@@ -152,4 +152,14 @@ export function log(msg, err) {
       console.log(logPrefix + msgLine + isoTime); // eslint-disable-line no-console
     }
   }
-}
\ No newline at end of file
+}
+export function getCloudDriveStatus() {
+  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/clouddrive/features/is-cloud-enabled`,{
```
Comment (azayati) [inline ecm-wcm-extension/src/main/webapp/attachments-selector/attachmentsService.js:157]:
```suggestion
  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/clouddrive/features/status`,{
```

File: apps/portlet-clouddrives/src/main/webapp/vue-app/cloudDriveService.js [hunk 2]
```diff
@@ -21,3 +21,14 @@ export async function getUserDrive() {
   }
 }
 
+export function getCloudDriveStatus() {
+  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/clouddrive/features/is-cloud-enabled`,{
```
Comment (azayati) [inline apps/portlet-clouddrives/src/main/webapp/vue-app/cloudDriveService.js:25]:
```suggestion
  return fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/clouddrive/features/status`,{
```

File: apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  created() {
+    getCloudDriveStatus().then(data => {
+      this.status = data.result === "true";
+    });
+  },
+  methods: {
+    changeActive() {
+      try {
+        fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/clouddrive/features/cloud-Status`, {
+          method: "POST",
```
Comment (azayati) [inline apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue:70]:
```suggestion
          method: "PATCH",
```

File: apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  },
+  created() {
+    getCloudDriveStatus().then(data => {
+      this.status = data.result === "true";
+    });
+  },
+  methods: {
+    changeActive() {
+      try {
+        fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/clouddrive/features/cloud-Status`, {
```
Comment (azayati) [inline apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue:69]:
```suggestion
        fetch(`${eXo.env.portal.context}/${eXo.env.portal.rest}/clouddrive/features/status`, {
```

File: core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java [hunk 5]
```diff
@@ -181,4 +181,30 @@ public Response isAutosyncEnabled(@Context UriInfo uriInfo,
...
+    return Response.ok().entity("{\"result\":\"" + this.CloudDriveEnabled + "\"}").build();
+  }
+
+  /**
+   * Update cloud Drive permissions.
+   *
+   * @param uriInfo the uri info
+   * @return the response
+   */
+  @POST
```
Comment (azayati) [inline core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java:203]:
```suggestion
  @PATCH
```

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: apps/portlet-clouddrives/src/main/resources/locale/navigation/group/platform/administrators_en.properties [hunk 6]
```diff
@@ -0,0 +1 @@
+cloudDrive.storage.page=Cloud Storage
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/resources/locale/navigation/group/platform/administrators_en.properties]: Done !

File: apps/portlet-clouddrives/src/main/webapp/WEB-INF/conf/clouddrive/portal/group/platform/administrators/pages.xml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        xsi:schemaLocation="http://www.gatein.org/xml/ns/gatein_objects_1_3 http://www.gatein.org/xml/ns/gatein_objects_1_3"
+        xmlns="http://www.gatein.org/xml/ns/gatein_objects_1_3">
+  <page>
+    <name>cloudStorage</name>
+    <title>Cloud Storage</title>
+    <access-permissions>manager:/platform/administrators</access-permissions>
+    <edit-permission>manager:/platform/administrators</edit-permission>
+    <portlet-application>
+      <portlet>
+        <application-ref>cloud-storage</application-ref>
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/WEB-INF/conf/clouddrive/portal/group/platform/administrators/pages.xml]: Done !

File: apps/portlet-clouddrives/src/main/webapp/WEB-INF/conf/clouddrive/portal/group/platform/administrators/pages.xml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    02110-1301 USA, or see the FSF site: http://www.fsf.org.
+-->
+<page-set
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.gatein.org/xml/ns/gatein_objects_1_3 http://www.gatein.org/xml/ns/gatein_objects_1_3"
+        xmlns="http://www.gatein.org/xml/ns/gatein_objects_1_3">
+  <page>
+    <name>cloudStorage</name>
+    <title>Cloud Storage</title>
+    <access-permissions>manager:/platform/administrators</access-permissions>
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/WEB-INF/conf/clouddrive/portal/group/platform/administrators/pages.xml]: Updated !

File: apps/portlet-clouddrives/src/main/webapp/WEB-INF/gatein-resources.xml [hunk 9]
```diff
@@ -142,6 +142,25 @@
       <module>cloudDriveUtils</module>
     </depends>
   </module>
+  <module>
+    <name>cloudStorage</name>
+    <script>
+      <minify>false</minify>
+      <path>/js/clouddrives-storage.bundle.js</path>
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/WEB-INF/gatein-resources.xml]: Done !

File: apps/portlet-clouddrives/src/main/webapp/WEB-INF/portlet.xml [hunk 10]
```diff
@@ -18,4 +18,26 @@
...
     </portlet-info>
   </portlet>
+
+  <portlet>
+    <portlet-name>cloudStoragePortlet</portlet-name>
+    <display-name xml:lang="EN">Cloud storage portlet</display-name>
+    <portlet-class>org.exoplatform.commons.api.portlet.GenericDispatchedViewPortlet</portlet-class>
+    <init-param>
+      <name>portlet-view-dispatched-file-path</name>
+      <value>/cloud-storage.jsp</value>
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/WEB-INF/portlet.xml]: Done !

File: apps/portlet-clouddrives/src/main/webapp/WEB-INF/portlet.xml [hunk 11]
```diff
@@ -18,4 +18,26 @@
...
+    </init-param>
+    <expiration-cache>-1</expiration-cache>
+    <cache-scope>PUBLIC</cache-scope>
+    <supports>
+      <mime-type>text/html</mime-type>
+    </supports>
+    <resource-bundle>locale.clouddrive.CloudDrive</resource-bundle>
+    <portlet-info>
+      <title>Cloud storage portlet for eXo Platform</title>
+      <short-title>Cloud Drive Admin</short-title>
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/WEB-INF/portlet.xml]: Done !

File: core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java [hunk 12]
```diff
@@ -181,4 +181,30 @@ public Response isAutosyncEnabled(@Context UriInfo uriInfo,
...
   }
 
+  /**
+   * Checks if is cloud drive enabled.
+   *
+   * @param uriInfo the uri info
+   * @return the response
+   */
+  @GET
+  @Path("/is-cloud-enabled/")
```
[the remark this answers is not in the document]
Comment (AzmiTouil) [reply core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java]: Done !

File: ecm-wcm-extension/src/main/webapp/attachments-selector/attachmentsService.js (hunk 1 above)
Comment (AzmiTouil) [reply ecm-wcm-extension/src/main/webapp/attachments-selector/attachmentsService.js]: Done !

File: apps/portlet-clouddrives/src/main/webapp/vue-app/cloudDriveService.js (hunk 2 above)
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/vue-app/cloudDriveService.js]: Done !

File: apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue (hunk 3 above)
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue]: Done !

File: apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue (hunk 4 above)
Comment (AzmiTouil) [reply apps/portlet-clouddrives/src/main/webapp/vue-app/cloudStorage/components/CloudStorage.vue]: Done !

File: core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java (hunk 5 above)
Comment (AzmiTouil) [reply core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java]: Done !

File: core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java [hunk 13]
```diff
@@ -181,4 +187,37 @@ public Response isAutosyncEnabled(@Context UriInfo uriInfo,
     }
   }
 
+  /**
+   * Checks if is cloud drive enabled.
```
Comment (azayati) [inline core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java:191]:
```suggestion
   * Checks if cloud drive is enabled.
```

File: core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java [hunk 14]
```diff
@@ -181,4 +187,37 @@ public Response isAutosyncEnabled(@Context UriInfo uriInfo,
...
+  @Path("/status")
+  @RolesAllowed("users")
+  public Response isCloudDriveEnabled(@Context UriInfo uriInfo) {
+    SettingValue<?> isCloudDriveEnabled = settingService.get(org.exoplatform.commons.api.settings.data.Context.GLOBAL.id("CLOUD_DRIVE"), Scope.GLOBAL.id("CLOUD_DRIVE"), "isCloudDriveEnabled");
+    Boolean enabled = isCloudDriveEnabled != null ? Boolean.valueOf(isCloudDriveEnabled.getValue().toString()) : true;
+    return Response.ok().entity("{\"result\":\"" + enabled + "\"}").build();
+  }
+
+  /**
+   * Update cloud Drive permissions.
```
Comment (azayati) [inline core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java:206]:
```suggestion
   * Set cloud Drive permissions.
```

File: core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java [hunk 15]
```diff
@@ -181,4 +187,37 @@ public Response isAutosyncEnabled(@Context UriInfo uriInfo,
...
+   *
+   * @param uriInfo the uri info
+   * @return the response
+   */
+  @PATCH
+  @Path("/status")
+  @RolesAllowed("administrators")
+  public Response changeCloudDriveStatus(@Context UriInfo uriInfo) {
```
Comment (azayati) [inline core/connector/src/main/java/org/exoplatform/ecm/connector/clouddrives/FeaturesService.java:214]:
```suggestion
  public Response setCloudDriveStatus(@Context UriInfo uriInfo) {
```

**Review by azayati — CHANGES_REQUESTED** [review]
see review

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**